k8s/cursor-initializator-elastic/cursor-initializator-elastic.py

```diff
@@ -0,0 +1,127 @@
+JOBS = [NODE_EXPORTER, HOST_NODE_EXPORTER]
+
+
+def parse():
+    parser = argparse.ArgumentParser(description="Script to initialize cursors for vector shards")
+    parser.add_argument("elastic_url", help="Elasticsearch url, i.e. http://localhost:9200")
+    parser.add_argument("node_filter", help="Node filter for current vector shard i.e. a.*")
+    parser.add_argument("output_dir", help="Path to which to initialize cursors")
+    parser.add_argument("--index-pattern", help="Index patterns to search for", dest="index_pattern", action="append")
+    parser.add_argument("--timeout", help="Timeout for elasticsearch requests", dest="timeout", default=300, type=int)
+    parser.add_argument("--suf", help="Optional suffix at the dir names", dest="suf", default="", type=str)
+    return parser.parse_args()
+
+
+def get_logger():
+    FORMAT = "[%(asctime)s] %(levelname)-8s %(message)s"
+    logging.basicConfig(format=FORMAT, level=logging.INFO)
+    return logging.getLogger(SCRIPT_NAME)
+
+
+def main():
+    args = parse()
+    logger = get_logger()
+
+    logger.info(f"Initializing elastic client with url: {args.elastic_url}")
+    client = Elasticsearch(args.elastic_url, request_timeout=args.timeout)
+
+    indices = [i for i in list(client.indices.get(index=args.index_pattern)) if not i.startswith(".")]
+
+    total_nodes = {}
+    for pattern in args.index_pattern:
+        if not any([re.match(pattern, i) for i in indices]):
+            logger.info(f"No indices found for pattern {pattern}")
+            continue
+
+        field = "ic_node"
+        if pattern.startswith("boundary"):
+            field = "_HOSTNAME"
+        nodes = client.search(
+            index=pattern, body={"aggs": {"patterns": {"terms": {"field": f"{field}.keyword", "size": 2_000}}}}
+        )["aggregations"]["patterns"]["buckets"]
+
+        nodes = [
+            i["key"]
+            for i in nodes
+            if re.match(args.node_filter, i["key"]) and not i["key"].startswith("guest") and not i["key"] == "localhost"
+        ]
+
+        if len(nodes) == 0:
+            logger.info(f"No nodes found for pattern {pattern} and filter {args.node_filter}")
+            continue
+
+        for job in JOBS:
+            body = {
+                "size": len(nodes),
+                "query": {
+                    "bool": {
+                        "minimum_should_match": 1,
+                        "must": [{"term": {"job.keyword": job}}],
+                        "should": {"regexp": {f"{field}.keyword": f"{args.node_filter}"}},
+                    }
+                },
+                "collapse": {
+                    "field": f"{field}.keyword",
+                    "inner_hits": {"name": "most_recent", "size": 1, "sort": [{"timestamp": "desc"}]},
+                },
+            }
+
+            last_logs = client.search(index=pattern, body=body)["hits"]["hits"]
+            # logger.info(f"Last logs: {json.dumps(last_logs, indent=4)}")
```

Comment:
do we need this?

Reply:
I left it there for testing, but in reality no :+1: